### ISSUE TITLE
[clang-tidy] bugprone-use-after-move when "taking"  class member's value

### ISSUE THREAD

**Issue by dmitrmax** [llvm/llvm-project#113306]
A false positive happens on the following code:

```
#include <string>
#include <iostream>

class A {
public:
    A(const std::string& str) : m_string(str), m_unmovedString(str) {}

    const std::string& getString() const& { return m_string; }
    std::string&& getString() && { return std::move(m_string); }

    const std::string& getUnmovedString() const { return m_unmovedString; }
private:
    std::string m_string;

    std::string m_unmovedString;
};

int main() {
    A a("test");

    std::cout << std::move(a).getString() << std::endl;
    std::cout << a.getUnmovedString() << std::endl;

    return 0;
}

```

Output looks like:
```
$ clang-tidy-19 main.cpp -checks=bugprone-use-after-move --
1 warning generated.
/tmp/tidy/main.cpp:22:18: warning: 'a' used after it was moved [bugprone-use-after-move]
   22 |     std::cout << a.getUnmovedString() << std::endl;
      |                  ^
/tmp/tidy/main.cpp:21:31: note: move occurred here
   21 |     std::cout << std::move(a).getString() << std::endl;
      |                               ^

```

**Comment by pkaff** [comment]
Are there any reasonable workarounds for this issue, or why hasn't this gained more traction?
